### DISCUSSION TITLE
Create page for offline-created projects

### DISCUSSION THREAD

**Discussion by LudwigHTW** [remsfal/remsfal-frontend#422] [Ideas]
## Info

Issue related to #308

With the extension to a Progressive Web App, it is possible to create projects offline, which are then sent to the backend as soon as the user is back online.

Currently, when a project is created offline, there is no visual feedback. The user does not receive any information about what happened after clicking "Create." This needs to be changed.

When a user creates a project offline, they should be redirected to a page that indicates they are offline and provides information about the offline created projects.

## Acceptance Criteria

AC 1: When a user creates a project offline, they are redirected to a new page called "OfflineProjects."
AC 2: The "OfflineProjects" page displays...
AC 2.1: a table, modeled after the "Übersicht der Liegenschaften" table
AC 2.1.1: with the title: "Übersicht der Offline erstellten Liegenschaften"
AC 2.1.2: with the description/text: “Du bist gerade offline. Wenn du wieder online bist, werden folgende Projekte erstellt.”
AC 2.1.3: and the columns: "Titel" and "Erstellt am"
AC 2.1.4: and the offline-created projects.
AC 3: When the projects are synced and therefore removed from the indexeddb, the table should be empty.
AC 4: The "OfflineProjects" page is stored in the cache.

Regarding AC 4: Since the Service Worker only caches pages visited by the user, the "OfflineProjects" page must be explicitly added to the cache.

## Dev Notes

The offline-stored projects are located in the indexeddb. For this, the function getAllProjects is available under helper/indexeddb.ts. When offline projects are synced they get deleted from the indexeddb (this part is already implemented).

**Comment by Eyad-Wahdan** [comment]
I think this ticket is a great idea! Perhaps the following points could be considered:

- Pluralization for Projects: Update the description and table to use "projects" instead of "project" to support scenarios where users create multiple projects offline. This will make the functionality more flexible and user-friendly.

- Handling Synchronized Projects: There is no mechanism or description of what happens to offline-created projects once they are synchronized. It would improve the user experience if the table automatically clears once the projects have been successfully sent to the backend. This would ensure consistency and clarity for the user.

**Comment by LudwigHTW** [comment]
good points, the generalization definitely makes sense. I have updated the issue accordingly.
